packages/jade-compiler/lib/transpilers.js

```diff
@@ -58,12 +121,40 @@ _.extend(FileCompiler.prototype, {
     for (var i = 0; i < self.nodes.length; i++)
       self.registerRootNode(self.nodes[i]);
 
-    return {
+    self.templatesHelpers=self.genTemplateHelpers(self.templates);
+    self.bodyHelpers=genHelpers(self.body);
+
+    r = {
       head: self.head,
       body: self.body,
       bodyAttrs: self.bodyAttrs,
-      templates: self.templates
+      templates: self.templates,
     };
+    // We actually wouldn't need bodyHelpers and templatesHelpers as
+    // there is a helpers attribute written in the root of the body
+    // and every template. Should we get rid of bodyHelpers and
+    // templatesHelpers (which is currently used by the file handler)?
+    if(!_.isEmpty(self.bodyHelpers))
+      r.bodyHelpers=self.bodyHelpers;
+    if(!_.isEmpty(self.templatesHelpers))
+      r.templatesHelpers=self.templatesHelpers
+    return r;
+  },
+
+  // Generate template helpers for every template.
+  genTemplateHelpers: function(templates) {
+    r={}
+    for (var key in templates) {
```

Comment:
Generally speaking `for x in y` should be prohibited because it leaks some keys from the prototypes. I’ve been using underscore helpers in this package (as does most packages in the Meteor ecosystem)